### ISSUE TITLE
about mapping software bwa!

### ISSUE THREAD

**Issue by sunnycqcn** [liangclab/HERA#7]
Hello,
Minimap2 usually should be much faster than bwa with long reads. Can we replace bwa with minimap2?
Thanks,
Fuyou

**Comment by chopra29** [comment]
It looks like a very useful tool but I think I will have to run step by step due to config of the server at my school. 

I have been trying to map with BWA it is loading corrected reads and contigs not scaffolded in RAM but after a few hours, there is no output.

I am wondering if bwa is not efficient at handling pacbio reads. So I wonder minimap2 will be a better option as well.

**Comment by liangclab** [comment]
Currently HERA does not work with Minimap2 directly because the output file format is different from that of BWA. BWA is slow but it outputs accurate overlap score which is required for HERA to distinguish repeats.

**Comment by chopra29** [comment]
Interesting, I am currently trying minimap2 and will post whatever I notice.

**Comment by acalcino** [comment]
I modified the script to run with minimap2 instead of bwa mem. I ran the test DJ assembly pipeline and here are the quast results:

Assembly	DJ_Final_Genome_HERA_bwa-mem	DJ_Final_Genome_HERA_minimap2
\# contigs (>= 0 bp)	49	49
\# contigs (>= 1000 bp)	49	49
\# contigs (>= 5000 bp)	49	49
\# contigs (>= 10000 bp)	49	49
\# contigs (>= 25000 bp)	14	14
\# contigs (>= 50000 bp)	1	1
Total length (>= 0 bp)	6021280	6021253
Total length (>= 1000 bp)	6021280	6021253
Total length (>= 5000 bp)	6021280	6021253
Total length (>= 10000 bp)	6021280	6021253
Total length (>= 25000 bp)	5375566	5375539
Total length (>= 50000 bp)	4977850	4977823
\# contigs	49	49
Largest contig	4977850	4977823
Total length	6021280	6021253
GC (%)	63.63	63.63
N50	4977850	4977823
N75	4977850	4977823
L50	1	1
L75	1	1
\# N's per 100 kbp	0.00	0.00

So in summary, the only difference to the assembly is 27nt but minimap2 is way faster and you can skip the whole bwa index step (perl $Working_Script/08-qsub_job_index.pl $genome_name $Sarray >>log). The minimap2 parameters I used were: 

minimap2 -ax asm20

Also, I modified the pipeline to run with Slurm hence the custom $Sarray variable there.

**Comment by sunnycqcn** [comment]
> I modified the script to run with minimap2 instead of bwa mem. I ran the test DJ assembly pipeline and here are the quast results:
> 
> Assembly DJ_Final_Genome_HERA_bwa-mem DJ_Final_Genome_HERA_minimap2
> # contigs (>= 0 bp) 49 49
> # contigs (>= 1000 bp) 49 49
> # contigs (>= 5000 bp) 49 49
> # contigs (>= 10000 bp) 49 49
> # contigs (>= 25000 bp) 14 14
> # contigs (>= 50000 bp) 1 1
> Total length (>= 0 bp) 6021280 6021253
> Total length (>= 1000 bp) 6021280 6021253
> Total length (>= 5000 bp) 6021280 6021253
> Total length (>= 10000 bp) 6021280 6021253
> Total length (>= 25000 bp) 5375566 5375539
> Total length (>= 50000 bp) 4977850 4977823
> # contigs 49 49
> Largest contig 4977850 4977823
> Total length 6021280 6021253
> GC (%) 63.63 63.63
> N50 4977850 4977823
> N75 4977850 4977823
> L50 1 1
> L75 1 1
> # N's per 100 kbp 0.00 0.00
> 
> So in summary, the only difference to the assembly is 27nt but minimap2 is way faster and you can skip the whole bwa index step (perl $Working_Script/08-qsub_job_index.pl $genome_name $Sarray >>log). The minimap2 parameters I used were:
> 
> minimap2 -ax asm20
> 
> Also, I modified the pipeline to run with Slurm hence the custom $Sarray variable there.

Hello,
Did you really finish the test of this pipeline? I find there should be have error in 21-Daligner_New.pl.
If you are done, could you share your experiences?
Thanks,
FUYOU

**Comment by acalcino** [comment]
Hey, late response but yes I did get it working. I modified 21-Daligner_New.pl to work on my local SLURM cluster but none of this matters because of the issues pointed out [here.](https://github.com/liangclab/HERA/issues/5) Until these are resolved I can't get HERA working on one of my real datasets anyway.
